contracts/riskon/UniswapV3RiskOnVault.sol

```diff
@@ -0,0 +1,673 @@
+        }
+    }
+
+    /// @notice Deposit to 3rd pool total assets
+    function depositTo3rdPoolTotalAssets() public view returns (uint256 _totalAssets) {
+        (uint256 _amount0, uint256 _amount1) = balanceOfPoolWants(baseMintInfo);
+        uint256 amount0 = _amount0;
+        uint256 amount1 = _amount1;
+        (_amount0, _amount1) = balanceOfPoolWants(limitMintInfo);
+        amount0 += _amount0;
+        amount1 += _amount1;
+        if (wantToken == token0()) {
+            _totalAssets = amount0 + uniswapV3RiskOnHelper.calcCanonicalAssetValue(token1(), amount1, token0());
+        } else {
+            _totalAssets = amount1 + uniswapV3RiskOnHelper.calcCanonicalAssetValue(token0(), amount0, token1());
+        }
+    }
+
+    /// @notice Gets the two tokens' balances of LP V3 NFT
+    /// @param _mintInfo  The info of LP V3 NFT
+    /// @return The amount of token0
+    /// @return The amount of token1
+    function balanceOfPoolWants(MintInfo memory _mintInfo) internal view returns (uint256, uint256) {
+        if (_mintInfo.tokenId == 0) return (0, 0);
+        (uint160 _sqrtPriceX96, , , , , ,) = pool.slot0();
+        return LiquidityAmounts.getAmountsForLiquidity(_sqrtPriceX96, TickMath.getSqrtRatioAtTick(_mintInfo.tickLower), TickMath.getSqrtRatioAtTick(_mintInfo.tickUpper), balanceOfLpToken(_mintInfo.tokenId));
+    }
+
+    /// @notice Harvests the Strategy
+    /// @return  _rewardsTokens The reward tokens list
+    /// @return _claimAmounts The claim amounts list
+    function harvest() public override returns (address[] memory _rewardsTokens, uint256[] memory _claimAmounts) {
+        lastHarvest = block.timestamp;
+        _rewardsTokens = new address[](2);
+        _rewardsTokens[0] = token0();
+        _rewardsTokens[1] = token1();
+        _claimAmounts = new uint256[](2);
+        uint256 _amount0;
+        uint256 _amount1;
+        if (baseMintInfo.tokenId > 0) {
+            (_amount0, _amount1) = __collectAll(baseMintInfo.tokenId);
+            _claimAmounts[0] += _amount0;
+            _claimAmounts[1] += _amount1;
+        }
+        if (limitMintInfo.tokenId > 0) {
+            (_amount0, _amount1) = __collectAll(limitMintInfo.tokenId);
+            _claimAmounts[0] += _amount0;
+            _claimAmounts[1] += _amount1;
+        }
+
+        if (profitFeeBps > 0 && address(treasury) != address(0)) {
+            if (_claimAmounts[0] > 0) {
+                uint256 claimAmount0Fee = _claimAmounts[0] * profitFeeBps / 10000;
+                _claimAmounts[0] -= claimAmount0Fee;
+                IERC20Upgradeable(token0()).safeApprove(address(treasury), 0);
+                IERC20Upgradeable(token0()).safeApprove(address(treasury), claimAmount0Fee);
```

Comment:
can put L209 and 210 into function initialize,approve type(uint256).max amount for gas save
```
 IERC20Upgradeable(token0()).safeApprove(address(treasury), type(uint256).max);
```